File: src/test/java/s5/TestS5.java
package s5;

import oopsla25.TestUtil;
import org.junit.Test;

public class TestS5 {

    @Test
    public void testFailure() {
        TestUtil.testGood(TestUtil.BASE_PATH + "\\s5\\good\\Failure.scr");
    }

    @Test
    public void testInterr() {
        TestUtil.testGood(TestUtil.BASE_PATH + "\\s5\\good\\Interr.scr");
    }

    @Test
    public void testAMQP() {
        TestUtil.testGood(TestUtil.BASE_PATH + "\\s5\\good\\AMQP.scr");
    }

    @Test
    public void testAMQP_long() {
        TestUtil.testGood(TestUtil.BASE_PATH + "\\s5\\good\\AMQP_long.scr");
    }


    /* Table 1 */

    @Test
    public void testCalculator() {
        TestUtil.testGood(TestUtil.BASE_PATH + "\\s5\\good\\table1\\Calculator.scr");
    }

    @Test
    public void testCircuitBreaker() {
        TestUtil.testGood(TestUtil.BASE_PATH + "\\s5\\good\\table1\\CircuitBreaker.scr");
    }

    @Test
    public void testDistributedLogging() {
        TestUtil.testGood(TestUtil.BASE_PATH + "\\s5\\good\\table1\\DistributedLogging.scr");
    }

    @Test
    public void testFibonacci() {
        TestUtil.testGood(TestUtil.BASE_PATH + "\\s5\\good\\table1\\Fibonacci.scr");
    }

    @Test
    public void testSMTP() {
        TestUtil.testGood(TestUtil.BASE_PATH + "\\s5\\good\\table1\\SMTP.scr");
    }

    @Test
    public void testTwoBuyer() {
        TestUtil.testGood(TestUtil.BASE_PATH + "\\s5\\good\\table1\\TwoBuyer.scr");
    }

    @Test
    public void testTravelAgency() {
        TestUtil.testGood(TestUtil.BASE_PATH + "\\s5\\good\\table1\\TravelAgency.scr");
    }

    @Test
    public void testOnlineWallet() {
        TestUtil.testGood(TestUtil.BASE_PATH + "\\s5\\good\\table1\\OnlineWallet.scr");
    }
}

File: src/test/scala/com/github/rhu1/gt/test/oopsla25/s5/TestS5.scala
package com.github.rhu1.gt.test.oopsla25.s5;

import com.github.rhu1.gt.test.oopsla25.TestUtil
import org.scalatest.funsuite.AnyFunSuite;


class TestS5 extends AnyFunSuite {

    test("Failure") {
        TestUtil.testGood(TestUtil.BASE_PATH + "\\s5\\good\\Failure.scr");
    }

    test("Interr") {
        TestUtil.testGood(TestUtil.BASE_PATH + "\\s5\\good\\Interr.scr");
    }

    test("AMQP") {
        TestUtil.testGood(TestUtil.BASE_PATH + "\\s5\\good\\AMQP.scr");
    }

    test("AMQP_long") {
        TestUtil.testGood(TestUtil.BASE_PATH + "\\s5\\good\\AMQP_long.scr");
    }


    /* Table 1 */

    test("Calculator") {
        TestUtil.testGood(TestUtil.BASE_PATH + "\\s5\\good\\table1\\Calculator.scr");
    }

    test("CircuitBreaker") {
        TestUtil.testGood(TestUtil.BASE_PATH + "\\s5\\good\\table1\\CircuitBreaker.scr");
    }

    test("DistibutedLogging") {
        TestUtil.testGood(TestUtil.BASE_PATH + "\\s5\\good\\table1\\DistributedLogging.scr");
    }

    test("Fibonacci") {
        TestUtil.testGood(TestUtil.BASE_PATH + "\\s5\\good\\table1\\Fibonacci.scr");
    }

    test("SMTP") {
        TestUtil.testGood(TestUtil.BASE_PATH + "\\s5\\good\\table1\\SMTP.scr");
    }

    test("TwoBuyer") {
        TestUtil.testGood(TestUtil.BASE_PATH + "\\s5\\good\\table1\\TwoBuyer.scr");
    }

    test("TravelAgency") {
        TestUtil.testGood(TestUtil.BASE_PATH + "\\s5\\good\\table1\\TravelAgency.scr");
    }

    test("OnlineWallet") {
        TestUtil.testGood(TestUtil.BASE_PATH + "\\s5\\good\\table1\\OnlineWallet.scr");
    }
}
